llvm/lib/Object/RecordStreamer.h

```diff
@@ -28,11 +29,11 @@ class RecordStreamer : public MCStreamer {
 
 private:
   const Module &M;
-  StringMap<State> Symbols;
+  MapVector<StringRef, State> Symbols;
   // Map of aliases created by .symver directives, saved so we can update
   // their symbol binding after parsing complete. This maps from each
   // aliasee to its list of aliases.
-  DenseMap<const MCSymbol *, std::vector<StringRef>> SymverAliasMap;
+  MapVector<const MCSymbol *, std::vector<StringRef>> SymverAliasMap;
```

Comment:
This one I see that it was non deterministic `DenseMap<const MCSymbol *`

Reply:
yes, this may cause nondeterminism problem.